cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -346,8 +349,11 @@ public void updateApp(HttpRequest request, HttpResponder responder,
       throw new BadRequestException("Request body is invalid json: " + e.getMessage());
     }
 
+    boolean updateSchedules =
+      appRequest.canUpdateSchedules() == null ? appUpdateSchedules : appRequest.canUpdateSchedules();
+
     try {
-      applicationLifecycleService.updateApp(appId, appRequest, createProgramTerminator());
+      applicationLifecycleService.updateApp(appId, appRequest, createProgramTerminator(), updateSchedules);
```

Comment:
If we make ApplicationLifecycleService responsible for reading Constants.AppFabric.APP_UPDATE_SCHEDULES and determining the appropriate value of whether to update schedule or not we will not need to pass this from here